src/styles/main.scss

```diff
@@ -1,3 +1,83 @@
+$gap_size: 1px;
+$cell_size: 100px;
+$col_count: 7;
+$days_count: 31;
+$week:
+  "mon" 0,
+  "tue" 1,
+  "wed" 2,
+  "thu" 3,
+  "fri" 4,
+  "sat" 5,
+  "sun" 6;
+
+html {
+  font-family: Arial, Helvetica, sans-serif;
+}
+
 body {
   margin: 0;
 }
+
+.main {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+
+  gap: $gap-size;
+  padding: 10px;
+  width: $col_count * ($cell_size + $gap_size);
+
+  @for $monthLength from 28 through 31 {
+    &--length_#{$monthLength} {
+      .calendar__day:nth-child(n + #{$monthLength + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  @each $name, $week_day in $week {
+    &--start-day_#{$name} .calendar__day:first-child {
+      margin-left: $week-day * ($cell_size + $gap_size);
+    }
+  }
+
+  &__day {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+
+    box-sizing: border-box;
+    height: $cell-size;
+    width: $cell-size;
+
+    font-size: 30px;
+
+    background-color: #eee;
+    border: 1px black solid;
+
+    cursor: pointer;
+
+    &:hover {
+      background-color: #ffc0cb;
+
+      transform: translateY(-20px);
+      transition: 0.5s;
```

Comment:
don't use it like that, because it's the same as transition: all 0.3s
It's too hard for the browser
Check this
https://github.com/mate-academy/layout_catalog_hovers/blob/master/checklist.md